### ISSUE TITLE
backend code imports masterd code

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#475]
Originally reported of Google Code with ID 419.

```
lib/backend.py imports mcpu.py, in order to have access to the HooksMaster class. However, this implies that cmdlib itself (and masterd/*) is imported, basically making noded use most of masterd's libraries.

This is not nice; it bumps noded memory usage by ~45% (18MB→26MB), and noded memory is locked in RAM, so not nice.

This was introduced in commit 0fa481f5, and went unseen for > 1 year now. I suspect simply moving HooksMaster to a separate file would be enough.
```
Originally added on 2013-04-09 14:24:39 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-04-10 18:15:05 +0000 UTC.

Added Labels: Priority-High
Added to Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-04-11 15:28:16 +0000 UTC.

Added Labels: Priority-Critical
Removed Labels: Priority-High

**Comment by Ganeti-Issues-Migrator** [comment]
```
I'll handle this one.
```
Originally added on 2013-04-13 14:41:49 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-04-19 12:47:10 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:49 +0000 UTC.

Changed State: Released
